test/session-pool.js

```diff
@@ -16,2022 +16,1575 @@
+            assert.strictEqual(sessionPool.options.concurrency, 10);
+            assert.strictEqual(sessionPool.options.maxWait, 50);
+            assert.strictEqual(sessionPool.options.idlesAfter, 50);
+            assert.strictEqual(sessionPool.options.max, 100);
+            assert.strictEqual(sessionPool.options.writes, 0.5);
+            assert.strictEqual(sessionPool.options.min, 0);
+            assert.strictEqual(sessionPool.options.maxReads, 79);
+            assert.strictEqual(sessionPool.options.maxWrites, 21);
+            assert.strictEqual(sessionPool.options.minReads, 0);
+            assert.strictEqual(sessionPool.options.minWrites, 0);
+          });
+        });
 
-    it('should create arrays for available and borrowed sessions', function() {
-      assert.deepEqual(sessionPool.reads_, []);
-      assert.deepEqual(sessionPool.writes_, []);
-      assert.deepEqual(sessionPool.borrowed_, []);
-    });
+        it('should override DEFAULT for maxReads', function() {
+          return sessionPool.close().then(function() {
+            sessionPool = new SessionPool(DATABASE, {
+              writes: 0.5,
+              maxReads: 40,
+            });
+            assert.strictEqual(sessionPool.options.acquireTimeout, Infinity);
+            assert.strictEqual(sessionPool.options.concurrency, 10);
+            assert.strictEqual(sessionPool.options.maxWait, 50);
+            assert.strictEqual(sessionPool.options.idlesAfter, 50);
+            assert.strictEqual(sessionPool.options.max, 100);
+            assert.strictEqual(sessionPool.options.writes, 0.5);
+            assert.strictEqual(sessionPool.options.min, 0);
+            assert.strictEqual(sessionPool.options.maxReads, 40);
+            assert.strictEqual(sessionPool.options.maxWrites, 60);
+            assert.strictEqual(sessionPool.options.minReads, 0);
+            assert.strictEqual(sessionPool.options.minWrites, 0);
+          });
+        });
 
-    it('should create an acquire queue with 1 concurrency', function() {
-      pQueueOverride = function(options) {
-        return options;
-      };
+        it('should override DEFAULT for minReads and minWrites', function() {
+          return sessionPool.close().then(function() {
+            sessionPool = new SessionPool(DATABASE, {
+              writes: 0.5,
+              maxReads: 40,
+              minReads: 41,
+              minWrites: 61,
+              min: 70,
+            });
+            assert.strictEqual(sessionPool.options.acquireTimeout, Infinity);
+            assert.strictEqual(sessionPool.options.concurrency, 10);
+            assert.strictEqual(sessionPool.options.maxWait, 50);
+            assert.strictEqual(sessionPool.options.idlesAfter, 50);
+            assert.strictEqual(sessionPool.options.max, 100);
+            assert.strictEqual(sessionPool.options.writes, 0.5);
+            assert.strictEqual(sessionPool.options.min, 70);
+            assert.strictEqual(sessionPool.options.maxReads, 40);
```

Comment:
I understand why we change the value, but I think it might be better to throw an error in the event the user provides us with a bunch of wonky values.